zebra/zebra_nhg.c

```diff
@@ -2523,10 +2530,24 @@ static int nexthop_active(struct nexthop *nexthop, struct nhg_hash_entry *nhe,
 						__func__, match, match->nhe,
 						newhop);
 
-				SET_FLAG(nexthop->flags,
-					 NEXTHOP_FLAG_RECURSIVE);
-				resolver = nexthop_set_resolved(afi, newhop,
+				SET_FLAG(nexthop->flags, NEXTHOP_FLAG_RECURSIVE);
+
+				if (policy && policy->status == ZEBRA_SR_POLICY_UP) {
+					if (policy->type == ZEBRA_SR_POLICY_TYPE_SRV6) {
+						SET_FLAG(nexthop->flags, NEXTHOP_FLAG_SRV6_TUNNEL);
+						resolver = nexthop_set_resolved(afi, newhop,
```

Comment:
this flag seems to be already set in nexthop_set_resolved(). not sure it is needed to set it before the functio.